locations/spiders/jimmy_johns.py

```diff
@@ -1,61 +1,53 @@
-          "SD", "TN", "TX", "UT", "VT", "VA", "WA", "WV", "WI", "WY"]
-HEADERS = { 'Content-Type': 'application/json' }
-JJBASE = 'https://www.jimmyjohns.com/webservices/Location/LocationServiceHandler.asmx/{}'
-CITIES = JJBASE.format('GetCitiesByStateNameAbbreviation')
-STORES = JJBASE.format('GetStoreAddressesByCityAndState')
+class TemplateSpider(scrapy.Spider):
+    name = "jimmy_johns"
+    allowed_domains = ["locations.jimmyjohns.com"]
+    start_urls = (
+        'https://locations.jimmyjohns.com/sitemap.xml',
+        #'https://locations.jimmyjohns.com/ks/emporia/sandwiches-1298.html',
+    )
+
+    def parse(self, response):
+        
+        stores = response.xpath('//url/loc[contains(text(),"sandwiches")]/text()').extract()
+        for store in stores:
+            print(store)
+            yield scrapy.Request(response.urljoin(store), callback=self.parse_store)
 
-class JimmyJohnsSpider(scrapy.Spider):
-    name = "jimmy-johns"
-    item_attributes = { 'brand': "Jimmy John's", 'brand_wikidata': "Q1689380" }
-    allowed_domains = ["www.jimmyjohns.com"]
-    download_delay = 0.2
+    def parse_store(self, response):
+        address1 = response.xpath('normalize-space(//p[@class="address"]/span[1]/text())').extract()
+        address2 = response.xpath('normalize-space(//p[@class="address"]/span[2]/text())').extract()
+        address2 = address2[0].split(",")
+        city = address2[0]
+        address2 = address2[1].strip()
+        address2 = address2.split(" ")
+        state = address2[0]
+        postcode = address2[1]
+        phone = response.xpath('//div[@class="map-list-item-inner relative"]/div/a[@class="phone ga-link"]/text()').extract()
+        gmapsurl = response.xpath('//div[@class="map-list-item-inner relative"]/a[@class="directions ga-link"]/@href').extract()
 
-    def start_requests(self):
-        for state in STATES:
-            current_state = json.dumps({ 'state': state })
-            request = scrapy.Request(
-                CITIES,
-                method='POST',
-                body=current_state,
-                headers=HEADERS,
-                callback=self.parse_cities
-            )
-            request.meta['state'] = state
-            yield request
+        _, query_string = parse.splitquery(gmapsurl[0])
+        query = parse.parse_qs(query_string)
+        latlng = query["daddr"]
+        lat, lon = latlng[0].split(",")
 
-    def parse_cities(self, response):
-        cities = json.loads(response.body)
-        for city in cities['d']:
-            current_city = json.dumps({ 'state': response.meta['state'], 'city': city })
-            request = scrapy.Request(
-                STORES,
-                method='POST',
-                body=current_city,
-                headers=HEADERS,
-                callback=self.parse
-            )
-            yield request
+        properties = {
+            'ref': response.url,
+            #'name': name,
+            'addr_full': address1[0],
```

Comment:
Instead of calling `extract()` and then taking the first index of the returned array, use `extract_first()`.